content/doc/developer/views/table-to-div-migration.adoc

```diff
@@ -0,0 +1,88 @@
+---
+title: Table to div layout migration
+layout: developerguide
+---
+
+Jenkins core has changed its form layout from 'table' to 'div' as of Jenkins v2.237.
+
+All core taglib's and views were updated in link:https://github.com/jenkinsci/jenkins/pull/3895[jenkins#3895]
+as part of jira:JENKINS-56109[].
+
+For 99% of plugins they do not require any changes at all as they just use the standard jelly tags
+in Jenkins for their form, such as `f:entry`, `f:textbox`, `f:checkbox`.
+
+== Examples of area which may require changes
+
+Below is some areas that may require you to adjust the form layout or JavaScript in your plugin:
+
+* Adding the td / tr elements directly instead of using the `f:entry` tag, e.g. link:https://github.com/jenkinsci/config-rotator-plugin/pull/2[config-rotator-plugin#2]
+
+* Custom javascript that is brittle to the Jenkins form UI layout, we’ve done our best to introduce shims, to not cause breakage here, 
+but if you have written JavaScript that is reliant on the DOM structure you may need to make adjustments,
+take a look at the JavaScript changes in link:https://github.com/jenkinsci/jenkins/pull/3895[jenkins#3895].
+
+* Forking jelly taglibs from core and extending them, i.e. link:https://github.com/jenkinsci/multiple-scms-plugin/pull/25[multiple-scms-plugin#25],
+it would be good to contribute fixes or enhancements to core rather than doing this where possible.
+
+* Using tables in the plugin for layout, link:https://github.com/jfrog/jenkins-artifactory-plugin/pull/266[jenkins-artifactory-plugin#266], link:https://github.com/jenkinsci/matrix-auth-plugin/pull/83[matrix-auth-plugin#83]
```

Comment:
```suggestion
* Using tables in the plugin for layout, link:https://github.com/jfrog/jenkins-artifactory-plugin/pull/266[jenkins-artifactory-plugin#266], link:https://github.com/jenkinsci/matrix-auth-plugin/pull/83[matrix-auth-plugin#83], ...
```

Reply:
not sure what you mean by this?

Reply:
It's a complaint about a fragment of the form `Using foo for blah, example1, example2` it isn't a sentence and as structured doesn't work. It might work as `Using foo for blah. example1, example2, ...`. There might be other ways to fix it.

Reply:
```suggestion
* Using tables in the plugin for layout, e.g. link:https://github.com/jfrog/jenkins-artifactory-plugin/pull/266[jenkins-artifactory-plugin#266] and link:https://github.com/jenkinsci/matrix-auth-plugin/pull/83[matrix-auth-plugin#83].
```